mypy/sametypes.py

```diff
@@ -153,14 +153,32 @@ def visit_literal_type(self, left: LiteralType) -> bool:
 
     def visit_union_type(self, left: UnionType) -> bool:
         if isinstance(self.right, UnionType):
+            # fast path for simple literals
+            def _extract_literals(u: UnionType) -> Tuple[Set[Type], List[Type]]:
```

Comment:
I believe nested functions don't perform well with mypyc, can you make this a global function instead?

Reply:
Done